apps/emqx_durable_storage/src/emqx_ds_buffer.erl

```diff
@@ -115,18 +133,22 @@ store_batch(DB, Messages, Opts) ->
     pending_replies = [] :: [gen_server:from()]
 }).
 
-init([DB, Shard]) ->
+init([CBM, CBMOptions, DB, Shard]) ->
     process_flag(trap_exit, true),
     process_flag(message_queue_data, off_heap),
-    logger:update_process_metadata(#{domain => [emqx, ds, egress, DB]}),
+    logger:update_process_metadata(#{domain => [emqx, ds, buffer, DB]}),
     MetricsId = emqx_ds_builtin_metrics:shard_metric_id(DB, Shard),
     ok = emqx_ds_builtin_metrics:init_for_shard(MetricsId),
+    {ok, CallbackS} = CBM:init_buffer(DB, Shard, CBMOptions),
     S = #s{
+        callback_module = CBM,
+        callback_state = CallbackS,
         db = DB,
         shard = Shard,
         metrics_id = MetricsId,
         queue = queue:new()
     },
+    persistent_term:put(?cbm(DB), {CBM, CBMOptions}),
```

Comment:
Nit. TBH having shard-level processes write to DB-level persistent term feels a bit awkward. I wonder why this module can't pull this CBM info from `emqx_ds` or `emqx_ds_sup`.

Reply:
`emqx_ds` ideally should not be aware of any specifics of the backend, including whether or not it uses the buffer.